Incorporated Feedback; some restructuring (#30)
riscv/riscv-trace-spec: exampleAlgorithm.tex

If there are no branches that need to be reported, packet format 2 is used.  

If there are branches to report, format 1 is used.  If branch prediction is supported and is enabled, 

> then there is a choice of whether to output a full branch map, or a count of correctly predicted branches.  

Reviewer: ```suggestion
then the encoder may choose whether to output a full branch map, or a count of correctly predicted branches.  
```
It should be clear who has that choice.
